lib/components/tag/tag.dart

```diff
@@ -0,0 +1,213 @@
+      ..classes.add('tag');
+
+    _tagText = SpanElement()
+      ..classes.add('tag-text')
+      ..dataset['placeholder'] = "untitled tag";
+    _tagActions = SpanElement()..classes.add('tag-actions');
+
+    _tagText.innerText = this._text;
+    if (suggested) {
+      var suggestedIcon = SpanElement()..className = 'fas fa-robot tag-suggested__icon';
+      _tagText.append(suggestedIcon);
+      _tagText.classes.toggle('tag-text--suggested', true);
+    }
+
+    updateStyle(tagStyle);
+
+    if (_selectable) {
+      _tagText.onClick.listen((_) {
+        onSelect();
+      });
+    }
+
+    if (_acceptable) {
+      _acceptButton = Button(ButtonType.confirm, onClick: (_) {
+        _acceptTag();
+      }).renderElement;
+      _tagActions.append(_acceptButton);
+    }
+
+    if (_editable) {
+      _editButton = Button(ButtonType.edit, onClick: (_) {
+        _makeEditable();
+      }).renderElement;
+      _tagActions.append(_editButton);
+
+      _confirmButton = Button(ButtonType.confirm, onClick: (_) {
+        _confirmEdit();
+      }).renderElement;
+
+      _cancelButton = Button(ButtonType.cancel, onClick: (_) {
+        _cancelEditing();
+      }).renderElement;
+
+      if (!disableDoubleClickEdit) {
+        _tagText.onDoubleClick.listen((_) {
+          _makeEditable();
+        });
+      }
+    }
+
+    if (_removable) {
+      _deleteButton = Button(ButtonType.remove, onClick: (_) {
+        _deleteTag();
+      }).renderElement;
+      _tagActions.append(_deleteButton);
+    }
+
+    if (editable && editableOnAdd) {
+      _makeEditable();
+    }
+
+    renderElement..append(_tagText)..append(_tagActions);
+  }
+
+  void _makeEditable() {
+    _tagText
+      ..contentEditable = "true"
+      ..focus()
+      ..onKeyDown.listen((event) {
+        if (event.keyCode == KeyCode.ENTER) {
+          event.preventDefault();
+          _confirmEdit();
+        }
+        if (event.keyCode == KeyCode.ESC) {
+          event.preventDefault();
+          _cancelEditing();
+        }
+      });
+    renderElement.classes.toggle("tag-text--editing", true);
+
+    _tagActions.children.clear();
+    _tagActions..append(_confirmButton)..append(_cancelButton);
+  }
+
+  void _cancelEditing() {
+    _resetActions();
+    _tagText.innerText = _text;
+  }
+
+  void _confirmEdit() {
+    _resetActions();
+    if (_text != _tagText.innerText) {
+      _text = _tagText.innerText;
+      onEdit(_text);
+    }
+  }
+
+  void _deleteTag() {
+    _resetActions();
+    onDelete();
+  }
+
+  void _acceptTag() {
+    _resetActions();
+    onAccept();
+  }
+
+  void _resetActions() {
+    renderElement.classes.toggle("tag-text--editing", false);
+    _tagText..contentEditable = "false";
+    _tagActions.children.clear();
+    if (_editable) {
+      _tagActions.append(_editButton);
+    }
+    if (_removable) {
+      _tagActions.append(_deleteButton);
+    }
+    if (_acceptable) {
+      _tagActions.append(_acceptButton);
+    }
```

Comment:
I think these should appear the other way around in the actions list (I see you're adding them in that order in the constructor though).

To avoid any confusions, can you also swap these two around in the class attributes list and constructor parameters?